### ISSUE TITLE
Allow LaunchType to be set per service, not per environment.

### ISSUE THREAD

**Issue by brentley** [stelligent/mu#265]
ECS clusters are heterogenous, so it would be cool if we could launch both fargate and ec2 based services into the same cluster. It looks like we export LaunchType from the environment and all services targeting that environment inherit that LaunchType.

If we move this decision into the service definition, then we can decide to launch both types in the same cluster.

**Comment by cplee** [comment]
If the decision is moved to the service definition, then you will always be provisioning ECS container instances (via ASG).  Sure, you could set the min/max/desired size of ASG to 0 but still seems clumsy for situations where you only want Fargate...a use case i hope to see more of.

I think it'd be interesting to allow an environment with ec2 launch type to allow a fargate service....but not the other way (Fargate launch type environment with EC2 service).

Does that seem correct?

**Comment by brentley** [comment]
Yeah, that sounds right... the use case you mentioned is exactly the situation I found myself in... I have a cluster running EC2 type services, and wanted to convert it to fargate. The easy way would be to convert service-by-service, but that doesn't appear to be possible. Trying to convert the whole environment caused an error (if I remember correctly).

It's an interesting scenario... maybe we need both options? To set the environment to a specific type, then also allow service-by-service overrides?

**Comment by cplee** [comment]
This would work fine, but i'd suggest using different environments (one set for Fargate and another for EC2).
